app/src/main/kotlin/com/mapbox/vision/teaser/MainActivity.kt

```diff
@@ -1,55 +1,531 @@
+                lp.marginStart = margin
+            }
+            lp.marginEnd = margin
+            image.layoutParams = lp
+            return image
+        }
+
+        fun LaneEdgeType.toDrawableId(isForward: Boolean = false): Int = when (this) {
+            LaneEdgeType.Curb -> if (isForward) {
+                R.drawable.ic_right_curb
+            } else {
+                R.drawable.ic_left_curb
+            }
+            LaneEdgeType.Construction -> TODO()
+            LaneEdgeType.MarkupDashed -> R.drawable.ic_half_lane
+            LaneEdgeType.MarkupDoubleSolid -> R.drawable.ic_separator_double_lane
+            LaneEdgeType.MarkupSolid -> R.drawable.ic_separator_lane
+            LaneEdgeType.Unknown -> R.drawable.ic_unknown_lane
+        }
+
+        fun LaneDirection.toDrawableId() = when (this) {
+            LaneDirection.Backward -> R.drawable.ic_arrow
+            LaneDirection.Forward -> R.drawable.ic_arrow_forward
+            LaneDirection.Reverse -> R.drawable.ic_arrow_reversed
+            LaneDirection.Unknown -> TODO()
+        }
+
+        lines_detections_container.removeAllViews()
+
+        for (index in roadDescription.lanes.indices) {
+            val lane = roadDescription.lanes[index]
+            val leftMarkingImageView = getImageView(index == 0)
+            leftMarkingImageView.setImageResource(lane.leftEdge.type.toDrawableId())
+            lines_detections_container.addView(leftMarkingImageView)
+
+            val directionImageView = getImageView()
+            if (index == roadDescription.currentLaneIndex) {
+                directionImageView.setImageResource(R.drawable.ic_blue_arrow)
+            } else {
+                directionImageView.setImageResource(lane.direction.toDrawableId())
+            }
+            lines_detections_container.addView(directionImageView)
+
+            if (index == roadDescription.lanes.lastIndex) {
+                val rightMarkingImageView = getImageView()
+                rightMarkingImageView.setImageResource(lane.rightEdge.type.toDrawableId(true))
+                lines_detections_container.addView(rightMarkingImageView)
+            }
+        }
+    }
+
+    private fun hideSignsContainer() {
+        sign_info_container.removeAllViews()
+        sign_info_container.hide()
+    }
+
+    private fun hideLineDetectionContainer() {
+        line_departure.hide()
+        lines_detections_container.removeAllViews()
+        lines_detections_container.hide()
+    }
+
+    private fun setVisionMode(mode: VisionMode) {
+        fps_performance_view.resetAverageFps()
+        soundsPlayer.stop()
+
+        back.show()
+        dashboard_container.hide()
+        safety_mode_container.hide()
+        hideLineDetectionContainer()
+        hideSignsContainer()
+
+        visionMode = mode
+        when (visionMode) {
+            VisionMode.Classification -> {
+                vision_view.visualizationMode = VisualizationMode.Clear
+                tracker = Tracker(5)
```

Comment:
is it possible to clear it instead of creating new one?

Reply:
It doesn't look critical, I don't see any advantage for now